johnnydep/lib.py

```diff
@@ -66,7 +68,13 @@ def __init__(self, req_string, parent=None, index_url=None, env=None, extra_inde
                 index_url=index_url,
                 env=env,
                 extra_index_url=extra_index_url,
+                ignore_errors=self.ignore_errors,
             )
+        error = self.metadata.get("error")
+        if error:
+            self.failed = True
+            self.error = error.split('\n')[-3]
```

Comment:
This magic number looks fragile - why storing the line at -3 index?  What's so special about this line, and can you add a proper parsing for it?